src/extension.ts

```diff
@@ -26,207 +31,225 @@ export class NewComment implements vscode.Comment {
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  public static disposeAllCommentThreads() {
+    console.log("dispoosing all comment threads");
+    for (const thread of commentThreadPool) {
+      thread.dispose();
+    }
+  }
 }
 
-export function activate(context: vscode.ExtensionContext) {
-  const commentController = vscode.comments.createCommentController( "virtual-comment-system", "Virtual Comment System");
+export async function activate(context: vscode.ExtensionContext) {
   context.subscriptions.push(commentController);
 
-  commentController.commentingRangeProvider = { provideCommentingRanges: (document: vscode.TextDocument) => {
+  commentController.commentingRangeProvider = {
+    provideCommentingRanges: (document: vscode.TextDocument) => {
       return [new vscode.Range(0, 0, document.lineCount - 1, 0)];
     },
   };
-
   commentController.options = {
     placeHolder: "Type a new comment or add a snippet",
     prompt: " ",
   };
+  NewComment.showCommentThread();
+  const codelensProvider = new CodelensProvider();
+  languages.registerCodeLensProvider({ scheme: "file" }, codelensProvider);
 
-  let commentProvider = showComments();
-
-  vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
-	commentProvider = showComments();
-  });
-
-  vscode.workspace.onDidChangeTextDocument((event) => {
-	//disable codelens - 2
-	commentProvider.dispose();
-	// const editedDocument = event.document;
-  // const editedLine = editedDocument.lineAt(event.contentChanges[0].range.start);
-  // const editedLineNumber = editedLine.lineNumber;
-	// // lineConfig.e/ditedLine = editedLineNumber+1;
-  // const str = editedDocument.lineAt(editedLineNumber).text;
-  // if (str !== "") {
-  //   // edit the str hash to btoa(str)
-  //   editLineText(editedLineNumber+1, str);
-  // }
+  vscode.window.tabGroups.onDidChangeTabs((tabChangeEvent: any) => {
+    NewComment.disposeAllCommentThreads();
+    NewComment.showCommentThread(tabChangeEvent?.changed[0]?.input.uri || undefined);
   });
 
   vscode.window.onDidChangeActiveTextEditor((event) => {
     if (!event) return;
-    const newStr = event.document.uri.path.split(config.folderName);
-    const newStr2 = path.join(newStr[0], config.folderName, ".docs", newStr[1]);
-    config.currentFilePath = event.document.uri.path;
-    config.commentJSONPath = (newStr2 + ".json");
-    config.document = vscode.window.activeTextEditor?.document;
-    config.folderName = vscode.workspace.workspaceFolders !== undefined? vscode.workspace.workspaceFolders[0].name : "";
-    config.fileName = newStr2.slice(newStr2.lastIndexOf('/')+1,newStr2.length);
-    commentProvider.dispose();
-    commentProvider = showComments();
+    if (event.document.uri.scheme === 'file' && currentActiveFile?.path !== event?.document.uri.path && !event.document.uri.path.includes('commentinput-1')) {
+      currentActiveFile = event?.document.uri;
+      console.log(event.selection.active.line);

```

Comment:
@sharath-kannan Could we remove console logs if not required?

Reply:
removed.